project1/application.py

```diff
@@ -111,8 +115,68 @@ def logout():
     return redirect(url_for('index'))
 
 
-@app.route("/home/<Username>")
-def userHome(Username):
-    if Username in session:
-        return render_template("userDetails.html", username=Username, message="Successfully logged in.", heading="Welcome back")
+@app.route("/home")
+def userHome():
+    if "username" in session:
+	username = session["username"]
+        return render_template("userDetails.html", username=username, message="Successfully logged in.", heading="Welcome back")
     return redirect(url_for('index'))
+
+
+@app.route("/search",methods=["POST","GET"])
+def search():
+	
+	#usera=session['uname']
+	if "username" in session:
+		username = session["username"]
+
+		if request.method == "GET":
+			return render_template("search.html")
+		else:
+			result = request.form.get("search")
+			result = '%'+result+'%'
+			search_result = Books.query.filter(or_(Books.tittle.ilike(result), Books.author.ilike(result), Books.isbn.ilike(result),Books.year.ilike(result))).all()
+			if len(search_result) == 0:
+				return render_template("search.html",message = "No records found")
+			return render_template("search.html", books=search_result)
+			#return "success"
+	else:
+		return redirect(url_for("register"))
+
+
+@app.route("/bookpage/<isbn>", methods = ["POST","GET"])
+def bookpage(isbn):
+	
+	book_isbn=isbn
+
+	if "username" in session:
+			username = session["username"]
+			user_reviews = reviewRate.query.filter_by(Isbn = book_isbn).all()
+			res = requests.get("https://www.goodreads.com/book/review_counts.json",
+                           params={"key": "fKvLN1nI7uY5XcyXi7VgvQ", "isbns": book_isbn})
+			if request.method == "GET":
+				revie = reviewRate.query.filter(reviewRate.Isbn.like(book_isbn),reviewRate.Username.like(username)).first()
+
+				#user_reviews = reviewRate.query.filter_by(isbn=bookisbn).all()
```

Comment:
remove commented code.

Reply:
I have removed search method and removed commented code also